runtime/queries/column_timeseries.go

```diff
@@ -0,0 +1,537 @@
+		}
+	}
+	result += " ) "
+
+	return result
+}
+
+func getFilterFromMetricsViewFilters(filters *runtimev1.MetricsViewRequestFilter) string {
+	includeFilters := getFilterFromDimensionValuesFilter(filters.Include, "", "OR")
+	excludeFilters := getFilterFromDimensionValuesFilter(filters.Exclude, "NOT", "AND")
+	if includeFilters != "" && excludeFilters != "" {
+		return " ( " + includeFilters + ") AND (" + excludeFilters + ")"
+	} else if includeFilters != "" {
+		return includeFilters
+	} else if excludeFilters != "" {
+		return excludeFilters
+	} else {
+		return ""
+	}
+}
+
+func getFallbackMeasureName(index int, sqlName string) string {
+	if sqlName == "" {
+		s := fmt.Sprintf("measure_%d", index)
+		return s
+	} else {
+		return sqlName
+	}
+}
+
+var countName = "count"
+
+func normaliseMeasures(measures []*runtimev1.GenerateTimeSeriesRequest_BasicMeasure) []*runtimev1.GenerateTimeSeriesRequest_BasicMeasure {
+	if len(measures) == 0 {
+		return []*runtimev1.GenerateTimeSeriesRequest_BasicMeasure{
+			{
+				Expression: "count(*)",
+				SqlName:    countName,
+				Id:         "",
+			},
+		}
+	}
+	for i, measure := range measures {
+		measure.SqlName = getFallbackMeasureName(i, measure.SqlName)
+	}
+	return measures
+}
+
+const IsoFormat string = "2006-01-02T15:04:05.000Z"
+
+func sMap(k string, v float64) map[string]float64 {
+	m := make(map[string]float64, 1)
+	m[k] = v
+	return m
+}
+
+func convertToDateTruncSpecifier(specifier runtimev1.TimeGrain) string {
+	switch specifier {
+	case runtimev1.TimeGrain_TIME_GRAIN_MILLISECOND:
+		return "MILLISECOND"
+	case runtimev1.TimeGrain_TIME_GRAIN_SECOND:
+		return "SECOND"
+	case runtimev1.TimeGrain_TIME_GRAIN_MINUTE:
+		return "MINUTE"
+	case runtimev1.TimeGrain_TIME_GRAIN_HOUR:
+		return "HOUR"
+	case runtimev1.TimeGrain_TIME_GRAIN_DAY:
+		return "DAY"
+	case runtimev1.TimeGrain_TIME_GRAIN_WEEK:
+		return "WEEK"
+	case runtimev1.TimeGrain_TIME_GRAIN_MONTH:
+		return "MONTH"
+	case runtimev1.TimeGrain_TIME_GRAIN_YEAR:
+		return "YEAR"
+	}
+	panic(fmt.Errorf("unconvertable time grain specifier: %v", specifier))
+}
+
+func convertRowsToTimeSeriesValues(rows *drivers.Result, rowLength int) ([]*runtimev1.TimeSeriesValue, error) {
+	results := make([]*runtimev1.TimeSeriesValue, 0)
+	defer rows.Close()
+	var converr error
+	for rows.Next() {
+		value := runtimev1.TimeSeriesValue{}
+		results = append(results, &value)
+		row := make(map[string]interface{}, rowLength)
+		err := rows.MapScan(row)
+		if err != nil {
+			return results, err
+		}
+		value.Ts = row["ts"].(time.Time).Format(IsoFormat)
+		delete(row, "ts")
+		value.Records = make(map[string]float64, len(row))
+		for k, v := range row {
+			switch x := v.(type) {
+			case int32:
+				value.Records[k] = float64(x)
+			case int64:
+				value.Records[k] = float64(x)
+			case float32:
+				value.Records[k] = float64(x)
+			case float64:
+				value.Records[k] = x
+			default:
+				return nil, fmt.Errorf("unknown type %T ", v)
+			}
```

Comment:
This cast doesn't feel safe, since any measure expression can be passed. For now, we can add `*big.Int` case and we'll probably be safe. But when merging this query with the metrics view code, this should be addressed more generally.

Reply:
I guess that requires changes on UI side.

Reply:
We would need to cast `*big.Int` to a float for UI (and result would not be exact if exceeding), but though it's bad practice, that's what frontend currently wants (we do it other places as well). 

There are also `int8`, `int16`, `uint8`, `uint16`, and maybe other measure result types I'm not thinking of.